### PR TITLE
=act Validate Actor class in Props.

### PR DESCRIPTION
**Pull request by He-Pin**
refs:https://github.com/akka/akka/issues/28812

### CODE REVIEW

File: akka-actor/src/main/scala/akka/actor/AbstractProps.scala [hunk 1]
```diff
@@ -20,9 +20,17 @@ private[akka] trait AbstractProps {
...
+  private[akka] def validate(clazz: Class[_]): Unit = {
     if (Modifier.isAbstract(clazz.getModifiers))
       throw new IllegalArgumentException(s"Actor class [${clazz.getName}] must not be abstract")
+    clazz match {
+      case _ if classOf[Actor].isAssignableFrom(clazz)                 =>
+      case _ if classOf[IndirectActorProducer].isAssignableFrom(clazz) =>
+      case _ =>
+        throw new IllegalArgumentException(
+          s"Actor class [${clazz.getName}] must be subClass of akka.actor.Actor or akka.actor.IndirectActorProducer.")
+    }
```
Comment (johanandren) [inline akka-actor/src/main/scala/akka/actor/AbstractProps.scala]: Make it an if else instead since it doesn't actually pattern match anything

**Review by johanandren — COMMENTED** [review]
Looks like a nice little improvement to me.

File: akka-actor/src/main/scala/akka/actor/AbstractProps.scala (hunk 1 above)
Comment (hepin1989) [reply akka-actor/src/main/scala/akka/actor/AbstractProps.scala]: yeap

**Review by johanandren — APPROVED** [review]

**Review by octonato — APPROVED** [review]
LGTM, thanks @hepin1989

**Pull request merged**